python/foglamp/services/core/server.py

```diff
@@ -306,6 +316,24 @@ def get_certificates(cls):
             await cls._configuration_manager.create_category(category, config, 'FogLAMP Admin and User REST API', True, display_name="Admin API")
             config = await cls._configuration_manager.get_category_all_items(category)
 
+            try:
+                cls.is_auth_required = True if config['authentication']['value'] == "mandatory" else False
+            except KeyError:
+                _logger.error("error in retrieving authentication info")
+                raise
+
+            try:
+                cls.auth_method = config['authMethod']['value']
+                if cls.is_auth_required:
+                    if cls.auth_method not in ['password', 'certificate']:
+                        raise ValueError("For authentication = mandatory, authMethod must be either password or certificate")
```

Comment:
if auth is required the  authMethod allows to restrict to use password or certificate or you can use any!

Hence this check is wrong. 
if any is the  auth_method (which is the default too) user should be able to login either by username + password Or certificate.

if auth_method == certificate: user can only use certificate to enter into the system
if auth_method == password: user can only use password to enter into the system

Reply:
Basically "authMethod" is for GUI use only - to direct to appropriate page/action based upon its value. Backend will act per whatever method is choosen by GUI to login.

Reply:
No GUI, will provide both option. 

if auth is mandatory, GUI will throw the user on login screen. and allow to choose any. if server allows only password or certificate only then user will be informed.

Reply:
Yes, same intent was meant by my comment. It is GUI which will decide how to login based upon "authentication" and "authMethod".